src/control/cmd/dmg/storage.go

```diff
@@ -26,15 +26,17 @@ package main
 import (
 	"github.com/daos-stack/daos/src/control/client"
 	ctlpb "github.com/daos-stack/daos/src/control/common/proto/ctl"
+	mgmtpb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
 	types "github.com/daos-stack/daos/src/control/common/storage"
 )
 
 // storageCmd is the struct representing the top-level storage subcommand.
 type storageCmd struct {
-	Prepare storagePrepareCmd `command:"prepare" alias:"p" description:"Prepare SCM and NVMe storage attached to remote servers."`
-	Scan    storageScanCmd    `command:"scan" alias:"s" description:"Scan SCM and NVMe storage attached to remote servers."`
-	Format  storageFormatCmd  `command:"format" alias:"f" description:"Format SCM and NVMe storage attached to remote servers."`
-	Query   storageQueryCmd   `command:"query" alias:"q" description:"Query storage commands, including raw NVMe SSD device health stats and internal blobstore health info."`
+	Prepare   storagePrepareCmd   `command:"prepare" alias:"p" description:"Prepare SCM and NVMe storage attached to remote servers."`
+	Scan      storageScanCmd      `command:"scan" alias:"s" description:"Scan SCM and NVMe storage attached to remote servers."`
+	Format    storageFormatCmd    `command:"format" alias:"f" description:"Format SCM and NVMe storage attached to remote servers."`
+	Query     storageQueryCmd     `command:"query" alias:"q" description:"Query storage commands, including raw NVMe SSD device health stats and internal blobstore health info."`
+	SetFaulty storageSetFaultyCmd `command:"setfaulty" alias:"sf" descrption:"Manually set the device state of an NVMe SSD to FAULTY."`
```

Comment:
are there other parameters we are going to want to set on devices? if so we probably want "storage set faulty" but nothing that needs to be done specifically in this PR.

Reply:
this is also slightly confusing given the scan and format subcommands apply to both SCM and NVMe, what about having this command executed by syntax "storage set nvme-faulty --uuid xx"? @kccain

Reply:
yes I like specifying the nvme-faulty. maybe we should sync next week and chat about options?

Reply:
set nvme-faulty seems good.
are all of the dmg storage commands clear to what types of devices they apply? a possible idea (would need to examine the full set of commands), use nvme in those cases specific to nvme, and otherwise the (sub) command applies to both scm and nvme.

Reply:
yes I think that works for the current set of commands, we should try to keep the tools wiki up-to-date so we have a canonical reference

Reply:
added a new ticket to track the work to update the naming of this command https://jira.hpdd.intel.com/browse/DAOS-3847